### ISSUE TITLE
Unvalidated ExtensionPoint.manifest entries

### ISSUE THREAD

**Issue by astrelsky** [NationalSecurityAgency/ghidra#6671]
**Describe the bug**
The entries in an `ExtensionPoint.manifest` file are not validated before being added to the buffer that is passed to `Pattern.compile`. Entries containing problematic regex characters such as `\`, `(`, `[`, etc. will cause Ghidra to throw an exception on startup.

**To Reproduce**
1. Open an existing extensions `ExtensionPoint.manifest` file and add `Invalid.Cl\a(s[s`.
2. Attempt to start Ghidra and see "AssertException - Unable to locate extension points!"

**Expected behavior**
Entries containing invalid characters should be logged and skipped.

**Environment (please complete the following information):**
 - Ghidra Version: 11.2_DEV

**Additional context**
I like breaking things

**Comment by ghidra1** [comment]
I would think failing hard is preferable since such bad strings should be addressed within source instead of ignoring.

**Comment by astrelsky** [comment]
> I would think failing hard is preferable since such bad strings should be corrected within source instead of ignoring.

Right. It should tell you what went wrong though instead of "Unable to locate extension points!'
